src/runmode-dpdk.c

```diff
@@ -0,0 +1,1274 @@
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetMulticast(iconf, entry_bool);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (ConfGetChildValueBoolWithDefault(
+                if_root, if_default, dpdk_yaml.checksum_checks, &entry_bool) != 1) {
+        SCLogError(
+                SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.checksum_checks);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetChecksumChecks(iconf, entry_bool);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (ConfGetChildValueBoolWithDefault(
+                if_root, if_default, dpdk_yaml.checksum_checks_offload, &entry_bool) != 1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing",
+                dpdk_yaml.checksum_checks_offload);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetChecksumOffload(iconf, entry_bool);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (ConfGetChildValueWithDefault(if_root, if_default, dpdk_yaml.copy_mode, &copy_mode_str) !=
+            1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.copy_mode);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    if (ConfGetChildValueWithDefault(if_root, if_default, dpdk_yaml.copy_iface, &copy_iface_str) !=
+            1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.copy_iface);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetCopyIfaceSettings(iconf, copy_iface_str, copy_mode_str);
+    if (retval < 0) {
+        return retval;
+    }
+
+    SCReturnInt(0);
+}
+DPDKIfaceConfig *ConfigParse(const char *iface)
+{
+    int retval;
+    DPDKIfaceConfig *iconf = NULL;
+    if (iface == NULL) {
+        SCLogError(SC_ERR_DPDK_CONF, "Provided iface is NULL");
+        SCReturnPtr(NULL, "void *");
+    }
+
+    ConfigInit(&iconf);
+    if (iconf == NULL) {
+        SCReturnPtr(NULL, "void *");
+    }
+    retval = ConfigLoad(iconf, iface);
+    if (retval < 0) {
+        iconf->DerefFunc(iconf);
+        SCReturnPtr(NULL, "void *");
+    }
+
+    return iconf;
+}
+
+void DeviceInitPortConf(const DPDKIfaceConfig *iconf, const struct rte_eth_dev_info *dev_info,
+        struct rte_eth_conf *port_conf)
+{
+    *port_conf = (struct rte_eth_conf){
+            .rxmode = {
+                    .mq_mode = ETH_MQ_RX_NONE,
+                    .max_rx_pkt_len = RTE_ETHER_MAX_LEN,
+                    .offloads = 0, // turn every offload off to prevent any packet modification
+            },
+            .txmode = {
+                    .mq_mode = ETH_MQ_TX_NONE,
+                    .offloads = 0,
+            },
+    };
+
+    // configure RX offloads
+    if (dev_info->rx_offload_capa & DEV_RX_OFFLOAD_RSS_HASH) {
+        if (iconf->nb_rx_queues > 1) {
```

Comment:
rework to check if the itnerface support multiple rx-queue and support RSS. then allow rss configuration

Reply:
RSS supported by interface is not just related to number of RX queue but also NIC functionality. please rework

Reply:
The checks are already in place. The check whether the number of configured queues is less or equal to the number of allowed NIC queues is prior to RSS setting. (`iconf->nb_rx_queues > dev_info.max_rx_queues`)

Then the check whether NIC supports RSS is on the line 921 `if (dev_info->rx_offload_capa & DEV_RX_OFFLOAD_RSS_HASH) {`